packages/framework/src/Model/PriceList/PriceListDataFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\PriceList;
+
+class PriceListDataFactory
+{
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\PriceList\PriceListData
+     */
+    protected function createInstance(): PriceListData
+    {
+        return new PriceListData();
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\PriceList\PriceListData
+     */
+    public function create(): PriceListData
+    {
+        $priceListData = $this->createInstance();
+        $this->fillNew($priceListData);
+
+        return $priceListData;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\PriceList\PriceListData $priceListData
+     */
+    protected function fillNew(PriceListData $priceListData): void
+    {
```

Comment:
I don't think we want to rely on the fact that this factory will be used only in the administration. If someone start to transfer the price lists, this may be unpleasant surprise